components/brave_wallet/renderer/js_cardano_provider.cc

```diff
@@ -0,0 +1,252 @@
+    return v8::Local<v8::Promise>();
+  }
+
+  auto global_context(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
+  auto context(v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+
+  cardano_provider_->Enable(base::BindOnce(
+      &JSCardanoProvider::OnEnableResponse, weak_ptr_factory_.GetWeakPtr(),
+      std::move(global_context), std::move(promise_resolver), isolate));
+
+  return resolver.ToLocalChecked()->GetPromise();
+}
+
+void JSCardanoProvider::OnEnableResponse(
+    v8::Global<v8::Context> global_context,
+    v8::Global<v8::Promise::Resolver> promise_resolver,
+    v8::Isolate* isolate,
+    const bool success) {
+  if (!render_frame()) {
+    return;
+  }
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context = global_context.Get(isolate);
+  v8::Context::Scope context_scope(context);
+  v8::MicrotasksScope microtasks(isolate, context->GetMicrotaskQueue(),
+                                 v8::MicrotasksScope::kDoNotRunMicrotasks);
+
+  v8::Local<v8::Value> result = v8::Boolean::New(isolate, success);
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
+  if (success) {
+    gin::Handle<JSCardanoWalletApi> wallet_api = gin::CreateHandle(
+        isolate, new JSCardanoWalletApi(base::PassKey<JSCardanoProvider>(),
+                                        global_context.Get(isolate), isolate,
+                                        render_frame()));
+    if (wallet_api.IsEmpty()) {
+      return;
+    }
+    v8::Local<v8::Value> wallet_api_value = wallet_api.ToV8();
+    v8::Local<v8::Object> wallet_api_object =
+        wallet_api_value->ToObject(context).ToLocalChecked();
+
+    // Non-function properties are readonly guaranteed by gin::Wrappable
+    for (const std::string& method :
+         {"getNetworkId", "getUsedAddresses", "getUnusedAddresses",
+          "getChangeAddress", "getRewardAddresses", "getUtxos", "getBalance",
+          "signTx", "signData", "submitTx", "getExtensions", "getCollateral"}) {
+      SetOwnPropertyWritable(context, wallet_api_object,
+                             gin::StringToV8(isolate, method), false);
+    }
+
+    std::ignore = resolver->Resolve(context, std::move(wallet_api_object));
+  } else {
+    std::ignore = resolver->Reject(context, result);
+  }
+}
+
+v8::Local<v8::Promise> JSCardanoProvider::IsEnabled(v8::Isolate* isolate) {
+  if (!EnsureConnected()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  v8::MaybeLocal<v8::Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (resolver.IsEmpty()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  auto global_context(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
```

Comment:
Let's avoid `ToLocalChecked` here and below (in favor of `ToLocal(...)`). It just crashes if something is wrong and is not quite trivial to prove why ToLocalChecked never fails in each particular case.